### PR TITLE
Integrates Arxiv search and summarisation as available action.

### PR DESCRIPTION
**Pull request by Koobah**
First ever pull request. Integrates Arxiv search and summarisation as available action for auto-gpt.

Based on https://github.com/DaltonPayne/arxiv-chatbot, just updated to GPT 3.5

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -9,4 +10,12 @@ requests
...
\ No newline at end of file
+# Googlesearch python seems to be a bit cursed, anyone good at fixing thigns like this?
+=======
+beautifulsoup4==4.12.0
+colorama==0.4.6
+googlesearch_python==1.1.0
+openai==0.27.0
+readability_lxml==0.8.1
+requests==2.28.2
+>>>>>>> origin/master
```
Comment (dukedorje) [inline requirements.txt:21]: merge conflict artifacts

File: .env.template
Comment (sberney) [inline .env.template:1]: Undelete -- this file must remain

File: .DS_Store
Comment (sberney) [inline .DS_Store:1]: Don't commit this -- and add .DS_Store to the project's .gitignore file.

File: .gitignore [hunk 2]
```diff
@@ -8,4 +8,11 @@ scripts/auto_gpt_workspace/*
 *.mpeg
 .env
 last_run_ai_settings.yaml
-outputs/*
\ No newline at end of file
+outputs/*
+/.idea/.gitignore
+/.idea/Auto-GPT-original.iml
```
Comment (sberney) [inline .gitignore:13]: You have a leading slash that makes this not work -- and you can use glob patterns. For example:

```
.idea/*
```

or just

```
idea/
```

File: .idea/Auto-GPT.iml
Comment (sberney) [inline .idea/Auto-GPT.iml:1]: You need to remove this and the other .idea files

File: scripts/chat.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def chat_with_ai(
         full_message_history,
         permanent_memory,
         token_limit,
-        debug=False):
+        debug=True):
```
Comment (sberney) [inline scripts/chat.py:34]: revert

File: scripts/commands.py [hunk 4]
```diff
@@ -10,21 +11,24 @@
 from execute_code import execute_python_file
 from json_parser import fix_and_parse_json
 from googlesearch import search
+
+import llm_utils
```
Comment (sberney) [inline scripts/commands.py:15]: Can you drop this line from this file, here?

File: scripts/commands.py [hunk 5]
```diff
@@ -10,21 +11,24 @@
...
+
 cfg = Config()
 
 
 def get_command(response):
     try:
         response_json = fix_and_parse_json(response)
-
```
Comment (sberney) [inline scripts/commands.py:19]: If I were you I would undo all of these whitespace changes for everyone's sake.

File: scripts/data.py [hunk 6]
```diff
@@ -2,17 +2,43 @@
 from pathlib import Path
 
 
+# def load_prompt():
```
Comment (sberney) [inline scripts/data.py:5]: Commented out code should be removed

File: scripts/data.py [hunk 7]
```diff
@@ -2,17 +2,43 @@
...
         file_dir = Path(os.path.dirname(os.path.realpath(__file__)))
         data_dir = file_dir / "data"
         prompt_file = data_dir / "prompt.txt"
-        # Load the promt from data/prompt.txt
+        # Load the prompt from data/prompt.txt
         with open(prompt_file, "r") as prompt_file:
             prompt = prompt_file.read()
 
+        # Load prompt commands from prompt_commands.txt
+        prompt_commands_file = data_dir / "prompt_commands.txt"
```
Comment (sberney) [inline scripts/data.py:31]: No need to change the way the prompt is stored to add arxiv support.

File: scripts/data/prompt.txt [hunk 8]
```diff
@@ -3,26 +3,7 @@ CONSTRAINTS:
...
-9. Delete GPT Agent: "delete_agent", args: "key": "<key>"
-10. Write to file: "write_to_file", args: "file": "<file>", "text": "<text>"
-11. Read file: "read_file", args: "file": "<file>"
-12. Append to file: "append_to_file", args: "file": "<file>", "text": "<text>"
-13. Delete file: "delete_file", args: "file": "<file>"
-14. Evaluate Code: "evaluate_code", args: "code": "<full _code_string>"
-15. Get Improved Code: "improve_code", args: "suggestions": "<list_of_suggestions>", "code": "<full_code_string>"
-16. Write Tests: "write_tests", args: "code": "<full_code_string>", "focus": "<list_of_focus_areas>"
-17. Execute Python File: "execute_python_file", args: "file": "<file>"
-18. Task Complete (Shutdown): "task_complete", args: "reason": "<reason>"
```
Comment (sberney) [inline scripts/data/prompt.txt:25]: I recommend adding your line here, and restoring the contents of this file:

18. Scientific Research: "arxiv_search", args: "input": "<search>"

**Review by sberney — COMMENTED** [review]
Congratulations on your first PR!

In order not to introduce complications with the primary author and with other contributors, it's important not to change code, formatting, or organization that isn't directly pertinent to your PR.

To get your PR seriously considered (and it's a good idea, so I think you should), you need to clean things up and get the changelog to look as minimal and professional as possible.

**Comment by Koobah** [comment]
Thanks for the feedback. I thought that I removed this PR. As I have never
done it before, I want to first learn how to work with Git and contribute
to such project so it’s productive for everyone.

I will clean it up, observe and contribute later :)

čt 6. 4. 2023 v 9:13 odesílatel Sam Berney ***@***.***>
napsal:

**Comment by R3dFruitRollUp** [comment]
Did we get arxiv implemented?

**Pull request closed without merging**